File: Algorithms/src/nattybear/level1/NumberStrToNumber.java
package nattybear.level1;

class NumberStrToNumber {
	public int solution(String s) {
		int size = s.length();
		int cursor = 0;
		StringBuilder builder = new StringBuilder();

		while (cursor < size) {
			char c = s.charAt(cursor++);
			if (c >= '0' && c <= '9') {
				builder.append(c);
			} else if (c == 'o') {
				builder.append('1');
				cursor += 2;
			} else if (c == 't') {
				if (s.charAt(cursor++) == 'w') {
					builder.append('2');
					cursor++;
				} else {
					builder.append('3');
					cursor += 3;
				}
			} else if (c == 'f') {
				if (s.charAt(cursor++) == 'o') {
					builder.append('4');
					cursor += 2;
				} else {
					builder.append('5');
					cursor += 2;
				}
			} else if (c == 's') {
				if (s.charAt(cursor++) == 'i') {
					builder.append('6');
					cursor++;
				} else {
					builder.append('7');
					cursor += 3;
				}
			} else if (c == 'e') {
				builder.append('8');
				cursor += 4;
			} else if (c == 'n') {
				builder.append('9');
				cursor += 3;
			} else if (c == 'z') {
				builder.append('0');
				cursor += 3;
			}
		}

		return Integer.parseInt(builder.toString());
	}
}

File: nattybear/level1/숫자-문자열과-영단어/NumberStrToNumber.js
function solution(s) {
  const size = s.length;

  let cursor = 0;
  let result = "";

  while (cursor < size) {
    let c = s[cursor++];

    if (c >= '0' && c <= '9') {
      result += c;
    }
    else if (c == 'o') {
      result += '1';
      cursor += 2;
    }
    else if (c == 't') {
      if (s[cursor++] == 'w') {
        result += '2';
        cursor++;
      }
      else {
        result += '3';
        cursor += 3;
      }
    }
    else if (c == 'f') {
      if (s[cursor++] == 'o') {
        result += '4';
        cursor += 2;
      }
      else {
        result += '5';
        cursor += 2;
      }
    }
    else if (c == 's') {
      if (s[cursor++] == 'i') {
        result += '6';
        cursor++;
      }
      else {
        result += '7';
        cursor += 3;
      }
    }
    else if (c == 'e') {
      result += '8';
      cursor += 4;
    }
    else if (c == 'n') {
      result += '9';
      cursor += 3;
    }
    else if (c == 'z') {
      result += '0';
      cursor += 3;
    }
  }

  return Number(result);
}
